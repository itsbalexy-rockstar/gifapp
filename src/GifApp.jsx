import React from "react";
import Navbar from "./components/Navbar";
import { useState } from "react";
import GifCollection from "./components/GifCollection";

const GifApp = ({defaultCategories = []}) => {
  const [categories, setCategories] = useState(defaultCategories);
  return (
    <>
      <Navbar setCategories={setCategories} />
      <ul>
        {categories.map((category, idx) => (
          <GifCollection key={idx} category={category} />
        ))}
      </ul>
    </>
  );
};

export default GifApp;
