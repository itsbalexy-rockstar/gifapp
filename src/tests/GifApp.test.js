import React from 'react';
import GifApp from '../GifApp'
const { shallow } = require("enzyme");

describe('Test in <GifApp />', () => {
    test('Should show the correctly component', () => {
      const wrapper = shallow(<GifApp />)
      expect(wrapper).toMatchSnapshot()
    });
    test('Should show a categories list', () => {
        const categories = ['One Punch', 'Hola', 'Rick and Morty']
        const wrapper = shallow(<GifApp defaultCategories={categories}/>)
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifCollection').length).toBe(categories.length)
      });
    
})