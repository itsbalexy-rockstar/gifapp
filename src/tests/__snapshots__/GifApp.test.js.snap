// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifApp /> Should show a categories list 1`] = `
<Fragment>
  <Navbar
    setCategories={[Function]}
  />
  <ul>
    <GifCollection
      category="One Punch"
      key="0"
    />
    <GifCollection
      category="Hola"
      key="1"
    />
    <GifCollection
      category="Rick and Morty"
      key="2"
    />
  </ul>
</Fragment>
`;

exports[`Test in <GifApp /> Should show the correctly component 1`] = `
<Fragment>
  <Navbar
    setCategories={[Function]}
  />
  <ul />
</Fragment>
`;
